code/game/objects/items/robot/robot_upgrades.dm

```diff
@@ -507,41 +452,30 @@
 	model_type = list(/obj/item/robot_model/medical)
 	model_flags = BORG_MODEL_MEDICAL
 
-/obj/item/borg/upgrade/defib/action(mob/living/silicon/robot/R, user = usr)
+	items_to_add = list(/obj/item/shockpaddles/cyborg)
+
+/obj/item/borg/upgrade/defib/action(mob/living/silicon/robot/borg, user = usr)
 	. = ..()
 	if(.)
-		var/obj/item/borg/upgrade/defib/U = locate() in R
-		if(U)
-			to_chat(user, span_warning("This unit is already equipped with a defibrillator module!")) //check to see if we already have this module
-			return FALSE
-		var/obj/item/borg/upgrade/defib/backpack/BP = locate() in R //If a full defib unit was used to upgrade prior, we can just pop it out now and replace
-		if(BP)
-			BP.deactivate(R, user)
-			to_chat(user, span_notice("You remove the defibrillator unit to make room for the compact upgrade."))
-		var/obj/item/shockpaddles/cyborg/S = new(R.model)
-		R.model.basic_modules += S
-		R.model.add_module(S, FALSE, TRUE)
-
-/obj/item/borg/upgrade/defib/deactivate(mob/living/silicon/robot/R, user = usr)
-	. = ..()
-	if (.)
-		var/obj/item/shockpaddles/cyborg/S = locate() in R.model
-		R.model.remove_module(S, TRUE)
+		var/obj/item/borg/upgrade/defib/backpack/defib_pack = locate() in borg //If a full defib unit was used to upgrade prior, we can just pop it out now and replace
+		if(defib_pack)
+			defib_pack.deactivate(borg, user)
+			to_chat(user, span_notice("The defibrillator pops out of the chassis as the compact upgrade installs."))
 
 ///A version of the above that also acts as a holder of an actual defibrillator item used in place of the upgrade chip.
 /obj/item/borg/upgrade/defib/backpack
 	var/obj/item/defibrillator/defib_instance
 
-/obj/item/borg/upgrade/defib/backpack/Initialize(mapload, obj/item/defibrillator/D)
+/obj/item/borg/upgrade/defib/backpack/Initialize(mapload, obj/item/defibrillator/defib)
 	. = ..()
-	if(!D)
-		D = new /obj/item/defibrillator
-	defib_instance = D
+	if(!defib)
```

Comment:
```suggestion
	if(isnull(defib))
```

`isnull()` is more explicit than `!` in this instance (and as a added perk, very slightly faster)